__tests__/components/AnonymousChat/child/ChatItemMyTextV2.test.tsx

```diff
@@ -1,34 +1,103 @@
+jest.spyOn(React, 'useRef').mockReturnValue({current: {close: jest.fn()}});
+
+const avatar = (
+  <FastImage
+    style={{height: 24, width: 24, borderRadius: 12}}
+    source={{uri: DEFAULT_PROFILE_PIC_PATH}}
+  />
+);
+
+let setReplyPreview;
+let contextValue;
+let timestamp;
+
+beforeEach(() => {
+  setReplyPreview = jest.fn();
+  timestamp = new Date().toISOString();
+  contextValue = {
+    chat: [{replyTarget: null}, setReplyPreview],
+    profile: [{myProfile: {}}]
+  };
+});
+
+afterEach(() => {
+  jest.restoreAllMocks();
+});
+
+const renderChatItemMyTextV2 = (isContinous = false) =>
+  render(
+    <Context.Provider value={contextValue}>
       <ChatItemMyTextV2
-        avatar="https://www.google.com"
+        avatar={avatar}
         username="username"
-        isContinuous={false}
+        isContinuous={isContinous}
         message="message"
-        time="time"
+        time={timestamp}
+        chatType={SIGNED}
+        messageType="regular"
+        data={{
+          id: 'id986',
+          user: {username: 'username'},
+          message: 'message',
+          message_type: 'regular',
+          updated_at: timestamp,
+          chatType: SIGNED
+        }}
       />
-    );
+    </Context.Provider>
+  );
 
-    expect(toJSON()).toMatchSnapshot();
+describe('TESTING ChatItemMyTextV2', () => {
+  it('should call setReplyPreview when swiped', () => {
+    const {getByTestId} = renderChatItemMyTextV2();
+    const expected = {
+      payload: {
+        id: 'id986',
+        message: 'message',
+        message_type: 'regular',
+        chatType: 'SIGNED',
+        updated_at: timestamp,
+        user: {username: 'username'}
+      },
+      type: 'SET_REPLY_TARGET'
+    };
+    act(() => {
+      fireEvent(getByTestId('swipeable'), 'onSwipeableOpen', 'left');
+    });
+    expect(setReplyPreview).toHaveBeenCalledWith(expected);
   });
 
-  it('RENDER continuous should match snapshot', () => {
-    const {toJSON} = render(
-      <ChatItemMyTextV2
-        avatar="https://www.google.com"
-        username="username"
-        isContinuous={true}
-        message="message"
-        time="time"
-      />
-    );
+  it('should display the username and time when isContinuous is false', () => {
+    const {getByText, getByTestId} = renderChatItemMyTextV2();
+    expect(getByText('username')).toBeTruthy();
+    expect(getByTestId('timestamp')).toBeTruthy();
+  });
+
+  it('should not display the username and time when isContinuous is true', () => {
+    const {queryByText} = renderChatItemMyTextV2(true);
+    expect(queryByText('username')).toBeNull();
+    expect(queryByText('time')).toBeNull();
+  });
+
+  it('should display the message', () => {
+    const {getByText} = renderChatItemMyTextV2();
+    expect(getByText('message')).toBeTruthy();
+  });
 
-    expect(toJSON()).toMatchSnapshot();
+  it('should match snapshot', () => {
+    const tree = renderChatItemMyTextV2().toJSON();
+    expect(tree).toMatchSnapshot();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new test cases added are comprehensive and cover various scenarios such as swipe actions, continuous message display, username and time visibility, and snapshot testing. The use of context and mock functions is appropriate for simulating the behavior of the component in different situations. However, there's a minor issue with the test case on line 88-92. It checks for the absence of 'time' text but it should check for the absence of timestamp instead.

```diff
-    expect(queryByText('time')).toBeNull();
+    expect(queryByText(timestamp)).toBeNull();
```
This change ensures that the test case correctly verifies whether the timestamp is not displayed when `isContinuous` is true.


<!-- This is an auto-generated comment by OSS CodeRabbit -->